### ISSUE TITLE
Cannot read properties of undefined (reading 'pageContext') - Question

### ISSUE THREAD

**Issue by sairao77** [pnp/pnpjs#2744]
### Category
- [ ] Enhancement
- [ ] Bug
- [ X] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [   PNP - 3.16.0 SPFx V1.17.4      ]

Please specify what version(s) of SharePoint you are targeting: [    Online     ]

### Expected / Desired Behavior / Question
For the below code I am expecting it delete add or delete list items, however I get an error

    import { Web } from '@pnp/sp/webs';
    import { SPFI, SPFx, spfi } from '@pnp/sp';
    import "@pnp/sp/items";
    import "@pnp/sp/site-users/web";
    import "@pnp/sp/lists";
    import "@pnp/sp/webs";

    const _listweb = Web('https://tenantid.sharepoint.com/sites/testsite').using(SPFx(this.context));
    const user = (await _listweb.currentUser()).Title;
    console.log(user) 
    const listdetails = await _listweb.lists.getByTitle("Test List").items();
    console.log(listdetails)
    await _listweb.lists.getByTitle("Test List").items.getById(1).delete().then(_ => {});
    await _listweb.lists.getByTitle("Test List").items.Add({
    Title : 'test'
    })

### Observed Behavior
the delete and add calls throw me a "Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'pageContext')"
I am able to display list items and current user why not add and delete using the same context, what am I missing? 

Thank you for your help in answering the question!

**Comment by bcameron1231** [comment]
Hi, I'm having trouble reproducing. Can you confirm that the code above is **Identical** to the code you're running? We've seen issues before when the queryable objects are being re-instantiated over and over.  There are a couple in our issues log that are similar. e.,g https://github.com/pnp/pnpjs/issues/2364

**Comment by patrick-rodgers** [comment]
Yes, at some point when this code runs `this.context` is undefined. Perhaps we need to start throwing a specific error there.

**Comment by HaHaBlah1** [comment]
>
